drop-transfer/src/storage_dispatch.rs

```diff
@@ -178,13 +180,14 @@ impl From<&crate::Event> for Event {
             crate::Event::RequestQueued(transfer) => Event::Pending {
                 transfer_info: transfer.storage_info(),
             },
-            crate::Event::FileDownloadStarted(transfer, file) => Event::Started {
-                transfer_type: TransferType::Incoming,
-                transfer_id: transfer.id(),
-                file_id: file.to_string(),
-            },
-            crate::Event::FileUploadStarted(transfer, file) => Event::Started {
-                transfer_type: TransferType::Outgoing,
+            crate::Event::FileDownloadStarted(transfer, file, base_dir) => {
+                Event::FileDownloadStarted {
+                    transfer_id: transfer.id(),
+                    file_id: file.to_string(),
+                    base_dir: base_dir.clone(),
```

Comment:
`
```suggestion
                    base_dir: base_dir,
```

Reply:
`base_dir` from the event is `&String`, can't not clone here

Reply:
No, if it's `&String` then it's fine. I thought it was `String`